### ISSUE TITLE
plugin overwrites auth details for https://index.docker.io/v1/ if no registryCredentials are defined

### ISSUE THREAD

**Issue by Jaxsun** [bmuschko/gradle-docker-plugin#985]
### Expected Behavior
Dockerhub credentials in `~/.docker/config.json` should be used when interacting with dockerhub without additional configuration.

### Current Behavior
Dockerhub credentials in `~/.docker/config.json` are read but discarded if there is no `registryCredentials` defined in the gradle build. This happens as the plugin generates a default `registryCredentials` configuration which has no username or password for dockerhub. This blank config then overwrites the config read from disk.

### Context
We thought we were authenticated against dockerhub, however we started getting rate limit errors which caused our builds to fail.

### Steps to Reproduce (for bugs)
execute `docker login`
have a gradle project with no `registryCredentials` defined
execute `gradle dockerImage`

### Workaround
define fake registryCredentials

```
docker {
    registryCredentials {
        url = 'fake'
    }
}
```

I had to debug the plugin to discover the precise behaviour.

**Comment by bmuschko** [comment]
What version of the plugin are you using?

**Comment by Jaxsun** [comment]
This was on 6.7.0

**Comment by bmuschko** [comment]
I thought we had a test case for this. Would you be interested in providing a PR with a test that demonstrates the incorrect behavior and maybe a fix as well?

**Comment by Jaxsun** [comment]
Looking at the test cases now all the tests run through `lookupAuthConfigWithDefaultAuthConfig` whereas the plugin will call `lookupAuthConfig` the big difference being that `lookupAuthConfig` takes a `DockerRegistryCredentials` input which is what is causing the trouble. So to me it really looks like there is no test coverage for how the plugin configuration interacts with the on disk Docker configuration/credentials.

Honestly I'm not sure what the right approach is to fix this since I lack context on this history of this behaviour. We could check whether blank credentials are present and then if on disk credentials are found use those instead. (And we could optionally only employ that if we are operating against the default registry URL, https://index.docker.io/v1/). The problem with such an approach is then you cannot force anonymous access via the `registryCredentials` config, and I'm not sure if that's a desired outcome or not.

If you can suggest a desired outcome I'd be happy to open a PR, though I am having issues where when I run the Gradle build I get test failures on DockerRegistryCredentialsTest (how relevant D:) about too many mock invocations. 

```
com.bmuschko.gradle.docker.internal.RegistryAuthLocatorTest > AuthLocator works using store FAILED
    org.spockframework.mock.TooManyInvocationsError at RegistryAuthLocatorTest.groovy:39

com.bmuschko.gradle.docker.internal.RegistryAuthLocatorTest > AuthLocator works using helper and empty auth SKIPPED
    org.spockframework.mock.TooManyInvocationsError at RegistryAuthLocatorTest.groovy:56

com.bmuschko.gradle.docker.internal.RegistryAuthLocatorTest > AuthLocator works when helper returns no server URL SKIPPED
    org.spockframework.mock.TooManyInvocationsError at RegistryAuthLocatorTest.groovy:74
```

I'm not really experienced with Spock, and it's been a while since I've done any Groovy development so any pointers on what might be causing my issue here would be appreciated.

**Comment by AndreyBar** [comment]
@Jaxsun @bmuschko hey guys. Any updates on this bug? We use your plugin in our company and also suffer from this bug. Any news is appreciated, thanks.

**Comment by bmuschko** [comment]
No news from my end. I am little surprised as I don't see the issue in the tests of this plugin or during regular use in projects. Maybe it's a specific use case I am not running into. A good start would be to add test cases. I am just swamped right now and need someone else to take charge without getting too involved.

**Comment by milespomeroy** [comment]
The workaround of using a 'fake' url isn't working for me. Still doesn't seem like the config.json file creds are being used. 

Debug shows a X-Registry-Auth header that I decoded as:

```json
{
  "username": null,
  "password": null,
  "email": null,
  "serveraddress": "fake",
  "auth": null,
  "registrytoken": null,
  "identitytoken": null,
  "stackOrchestrator": null
}
```

**Comment by Sineaggi** [comment]
We encountered this issue today, I've created a fix (with a test) in #1065

**Comment by bmuschko** [comment]
This has been released.
